jina/drivers/__init__.py

```diff
@@ -187,13 +173,16 @@ def req(self) -> 'LazyRequest':
     @property
     def partial_reqs(self) -> Sequence['LazyRequest']:
         """The collected partial requests under the current ``request_id`` """
-        if self.expect_parts <= 1:
-            raise ValueError(
-                f'trying to get collected requests even though expect_parts={self.expect_parts},'
-                f'maybe you want to use self.req instead?'
-            )
+        if self.expect_parts > 1:
```

Comment:
is it always there this `attribute`? It is not set in `__init__`?

Reply:
no need to set
```
    @property
    def expect_parts(self) -> int:
        """The expected number of partial messages before trigger :meth:`handle` """
        return self.args.num_part if self.message.is_data_request else 1

```